src/modelmanager.cpp

```diff
@@ -828,6 +839,52 @@ void ModelManager::watcher(std::future<void> exitSignal) {
     SPDLOG_LOGGER_INFO(modelmanager_logger, "Stopped model manager thread");
 }
 
+void ModelManager::cleanerRoutine(uint32_t resourcesCleanupIntervalSec, uint32_t sequenceCleanerIntervalMinutes, std::future<void> cleanerExitSignal) {
+    SPDLOG_LOGGER_INFO(modelmanager_logger, "Started cleaner thread");
+
+    uint32_t resourcesCleanupIntervalMiliseconds = resourcesCleanupIntervalSec * 1000;
+    uint32_t sequenceCleanerIntervalMiliseconds = sequenceCleanerIntervalMinutes * 60 * 1000;
+
+    FunctorResourcesCleaner functorResourcesCleaner{*this};
+    FunctorSequenceCleaner functorSequenceCleaner{globalSequencesViewer};
+
+    ovms::cleanerRoutine(resourcesCleanupIntervalMiliseconds, functorResourcesCleaner, sequenceCleanerIntervalMiliseconds, functorSequenceCleaner, cleanerExitSignal);
+
+    SPDLOG_LOGGER_INFO(modelmanager_logger, "Stopped cleaner thread");
+}
+
+void cleanerRoutine(uint32_t resourcesCleanupInterval, FunctorResourcesCleaner& functorResourcesCleaner, uint32_t sequenceCleanerInterval, FunctorSequenceCleaner& functorSequenceCleaner, std::future<void>& cleanerExitSignal) {
+    uint32_t currentResourcesWaitTime = resourcesCleanupInterval;
+    uint32_t currentSequenceWaitTime = sequenceCleanerInterval;
+    bool shouldCheckForSequenceCleanup = sequenceCleanerInterval != 0;
+    uint32_t currentWaitTime = (!shouldCheckForSequenceCleanup || currentResourcesWaitTime < currentSequenceWaitTime) ? currentResourcesWaitTime : currentSequenceWaitTime;
```

Comment:
Correct me if I'm wrong, but the cleaner routine interval is set to the lowest value between ResourcesWaitTime and SequenceWaitTime. For some configurations this may be good (when (higher_wait_time % lower_wait_time) is equal to 0) but if not, the higher wait time can get far off.

Example: SequenceWaitTime=121s, ResourcesWaitTime=120s.

In this example, first sequence cleaner will get executed after 240s instead of 121s.
What about having wait time static - 1s and constantly check if it is already time to fire any cleaner?

@atobiszei @mkuczyns11

Reply:
currentWaitTime works in other way. With your example it would be like this:
CleanerThread will wait 120s and then cleanup resources. Then
currentWaitTime = 1s, sequenceWaitTime = 1s, ResourcesWaitTime = 120s.
After 1s sequence is cleared and waitTime values will look like this:
currentWaitTime = 119s, sequenceWaitTime = 121s, ResourcesWaitTime = 119s.

Reply:
Your suggestion most certainly would make the code easier to read and understand. I see one issue with tests. Currently we are testing 5 scenarios. Each scenario then would be extended to at least 2 seconds, since we are waiting 2 times to check the cleanup.